feat: Experiment Center (#1332)
auth0/docs-v2: main/docs/customize/experiment-center/integrations/page-templates-integration-guide.mdx

---
title: "Experiment Center and Page Templates Integration"
description: "How experiment context flows through Auth0 Universal Login page templates and partials, and how to branch template rendering based on variation."
sidebarTitle: Page Templates Integration
---

import { ReleaseStageNotice } from "/snippets/ReleaseStageNotice.jsx"

<ReleaseStageNotice
    feature="Auth0 Experiment Center"
    stage="beta"
    terms="true"
    contact="Auth0 Support"
/>

> When an experiment is active and a variation is assigned, the Experiment Center injects experiment context into the [page template](/docs/customize/login-pages/universal-login/customize-templates#customize-universal-login-page-templates) rendering context automatically. Page templates do not require an opt-in step: experiment context is passed to all templates unconditionally when an experiment is active. Templates receives the context as a local variable named `experiment`.

Reviewer: ```suggestion
When an experiment is active and a variation is assigned, the Experiment Center injects experiment context into the [page template](/docs/customize/login-pages/universal-login/customize-templates#customize-universal-login-page-templates), rendering context automatically. Page templates do not require an opt-in step: experiment context is passed to all templates unconditionally when an experiment is active. Templates receives the context as a local variable named `experiment`.
```

<Warning>
During Beta, Experiment Center runs on development tenants only. Production tenants are not supported.
</Warning>

## Available context fields

> The following fields are available in the template rendering context when an experiment is active:

Reviewer: ```suggestion
The following fields are available in the page template when an experiment is active:
```

> | Variable | Type | Description |

Reviewer: ```suggestion
| **Variable** | **Type** | **Description** |
```
